state/machine.go

```diff
@@ -1847,6 +1847,20 @@ func isSupportedContainer(container instance.ContainerType, supportedContainers
 
 // updateSupportedContainers sets the supported containers on this host machine.
 func (m *Machine) updateSupportedContainers(supportedContainers []instance.ContainerType) (err error) {
+	if m.doc.SupportedContainersKnown {
+		if len(m.doc.SupportedContainers) == len(supportedContainers) {
+			equal := true
+			for i := range m.doc.SupportedContainers {
+				if m.doc.SupportedContainers[i] != supportedContainers[i] {
```

Comment:
What happens if the order of supported containers is different, yet they are the same (can that even happen)?

Example:
```A, B, C -> C, A, B```